packages/itwinui-react/src/core/Menu/Menu.tsx

```diff
@@ -89,6 +89,16 @@ export const Menu = React.forwardRef((props, ref) => {
     }
   };
 
+  // Keep focusedIndex in sync with the real focused node's index.
+  const onFocus = (event: React.FocusEvent<HTMLDivElement, Element>) => {
+    const items = getFocusableNodes();
+    const currentIndex = items.findIndex((el) => el === event.target);
```

Comment:
Running `getFocusableNodes` and `findIndex` every time focus moves is expensive.

Suggestion to improve performance:
1. In `MenuItem`, set `data-iui-index={indexInParentTree}`.
2. In this function, read `e.target.dataset.index`.

Reply:
I believe I had made this change before using [`useListNavigation`](https://floating-ui.com/docs/useListNavigation). Back then, I was instead relying on `Menu` for the <kbd>↑</kbd>/<kbd>↓</kbd> arrow keys navigation.

But since I'm now using `useListNavigation` for keyboard navigation, the code changes in `Menu.tsx` are no longer necessary. So, I reverted all changes in `Menu.tsx` (https://github.com/iTwin/iTwinUI/pull/1942/commits/ffb4fa269f78a91bb87a21047e2156009ba4971d).